scripts/Muon/GUI/ElementalAnalysis2/grouping_widget/ea_grouping_table_widget_presenter.py

```diff
@@ -0,0 +1,160 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import re
+from Muon.GUI.Common.utilities import run_string_utils as run_utils
+from Muon.GUI.ElementalAnalysis2.ea_group import EAGroup
+from mantidqt.utils.observer_pattern import GenericObservable
+from Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_table_widget_view import inverse_group_table_columns
+
+
+maximum_number_of_groups = 20
+
+
+class EAGroupingTablePresenter(object):
+
+    def __init__(self, view, model):
+        self._view = view
+        self._model = model
+
+        self._view.on_remove_group_button_clicked(self.handle_remove_group_button_clicked)
+
+        self._view.on_user_changes_group_name(self.validate_group_name)
+
+        self._view.on_table_data_changed(self.handle_data_change)
+
+        self.selected_group_changed_notifier = GenericObservable()
+
+        self._dataChangedNotifier = lambda: 0
+
+    def on_data_changed(self, notifier):
```

Comment:
why do we need this?

Reply:
This is something that has come over from the Muon grouping_table_widget_presenter so not sure why we need it.

Reply:
lets delete it until we need it